### PR TITLE
adding linux-bridge bump script

### PR DESCRIPTION
**Pull request by RamLavi**
**What this PR does / why we need it**:
Currently cnao only bumps linux-bridge manually.
Let's add a script to do it for us.
This PR is a subset of an older PR #363 

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by oshoval** [comment]
```
In order to bump it, we add a script that will create a container with the appropriate linux-bridge tagged image, and push it to the registry for local use.
```
long line in description, please try to use max 72 char in body

File: hack/components/bump_linux_bridge.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+LINUX_BRIDGE_PATH=${GOPATH}/src/${LINUX_BRIDGE_REPO}
+
+echo 'Fetch linux-bridge sources'
+fetch_component ${LINUX_BRIDGE_PATH} ${LINUX_BRIDGE_URL} ${LINUX_BRIDGE_COMMIT}
+
+echo 'Get linux-bridge tag'
+LINUX_BRIDGE_TAG=$(get_component_tag ${LINUX_BRIDGE_PATH})
+
+echo 'Build container image with linux-bridge binaries'
+LINUX_BRIDGE_IMAGE=quay.io/kubevirt/cni-default-plugins
```
Comment (oshoval) [inline hack/components/bump_linux_bridge.sh:19]: generally speaking, maybe this worth to be a variable with default value, so one can override it, and bump the results also to his personal repository (need to be parameterized as well then of course), in order to create a custom bundle?

**Comment by RamLavi** [comment]
/hold
awaits merge of #448 and #449

File: hack/components/bump_linux_bridge.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+LINUX_BRIDGE_IMAGE_TAGGED=${LINUX_BRIDGE_IMAGE}:${LINUX_BRIDGE_TAG}
+(
+    cd ${LINUX_BRIDGE_PATH}
+    cat <<EOF > Dockerfile
+FROM golang:1.13 AS builder
```
Comment (qinqon) [inline hack/components/bump_linux_bridge.sh:24]: Building and pushing linux-bridge at CNAO feels strange, how were we doing it? maybe it deserve separate repo.

Comment (phoracek) [reply hack/components/bump_linux_bridge.sh:24]: We were doing it manually until now.

Separate repository would be a pain to maintain. I'd prefer to keep it here for our own sanity.

I agree that it is weird to build it in the same script we are bumping it. Would `build_linux_bridge.sh` make more sense? I would still call it from this bump script though.

We are consuming U/S components. For some we need to build manifests, for other we need to build whole images.

File: hack/components/bump_linux_bridge.sh (hunk 1 above)
Comment (RamLavi) [reply hack/components/bump_linux_bridge.sh]: IIUC there is only 1 image path at a certain point of time, and if it does changes ( like recently changed on multus) then it  should be changed in a proper PR. So I don't see a reason to make it a parameter that is changable from the outside

I mean if it was a common practice to use our own private images then maybe there would be a case for it, but from what I understood so far it's uncommon and not recommended

File: hack/components/bump-linux-bridge.sh [hunk 3]
```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+
+set -xe
```
Comment (oshoval) [inline hack/components/bump-linux-bridge.sh:3]: others have -o pipefail as well if you want to add here as well

Comment (ormergi) [reply hack/components/bump-linux-bridge.sh:3]: I agree, we should keep all bump scripts similar as possible for better maintenance and readability

File: hack/components/bump_linux_bridge.sh (hunk 2 above)
Comment (ormergi) [reply hack/components/bump_linux_bridge.sh]: How come there is no built images for CNI plugins like other components?
Should we raise it in containernetworking/plugin repository?

Comment (phoracek) [reply hack/components/bump_linux_bridge.sh]: Images as a delivery mechanism is a Kubernetes thing. containernetworking/plugins is a generic CNI project that can be used locally with podman etc.

**Pull request merged**